### PR TITLE
Add comments for documentation

### PR DESCRIPTION
**Pull request by ba58smith**
Reviewing @joelkoz's recent change to MovingAverage, I realized I didn't understand how it worked. The PR adds a description for the class documentation, and explains how the transform works.

### CODE REVIEW

File: src/transforms/moving_average.h [hunk 1]
```diff
@@ -5,12 +5,35 @@
...
 
+/**
+ * MovingAverage is used to smooth the output of a value (signal) that has frequent
+ * variations. For example, the output of a temperature sensor may vary from 180 to 185
+ * several times over a short period, but you just want to see the average of that.
+ * MovingAverage outputs the average of the most recent n values. It also incorporates
+ * a "scale" factor, in case you want to increase or decrease your final output by a
+ * fixed percentage.
+ * 
+ * @param n is the number of samples to average
```
Comment (joelkoz) [inline src/transforms/moving_average.h]: I think @param tag is only valid for comments that are on functions or the constructor.  Move these comments to just above the constructor declaration.

File: src/transforms/moving_average.h [hunk 2]
```diff
@@ -5,12 +5,35 @@
...
   MovingAverage(int n, float k=1., String config_path="");
+
+  /**
+   * set_input is used to input a new value to this transform. After the
+   * transformation, notify() is called to output the new value.
+   */
   virtual void set_input(float input, uint8_t inputChannel = 0) override;
+
+  /** 
+   * For reading and writing the configuration of this transform
```
Comment (joelkoz) [inline src/transforms/moving_average.h]: The way Doxygen works, this comment, which applies to the three methods below, will actually only apply to the first method declaration (get_configuration()). I would put NO comments on ANY of these methods, as they are all "virtual" and "override."  Instead, if you document each method in the base class where they are inherited from (if they are not already documented), then these overrides will "inherit" the documentation.  So, in general, we should not have any Doxygen docs for any "virtual ... override" methods except for the original declarations in the base class.  Conversely, we should try our best to ALWAYS document "virtual" methods declared in a base class so as to allow descendant classes to inherit those docs.

File: src/transforms/moving_average.h (hunk 1 above)
Comment (ba58smith) [reply src/transforms/moving_average.h]: Done.

File: src/transforms/moving_average.h (hunk 2 above)
Comment (ba58smith) [reply src/transforms/moving_average.h]: The base class is Configurable, and configurable.h has a comment for each of these methods, so I'll remove them from here. Thanks for the clarification. (I was using curveinterpolator.h as a model for doing this - it has "//" in the places where I put the "/**" comments.)

**Comment by mairas-cujo** [comment]
Sorry, forgot about this!

**Pull request merged**